### PR TITLE
Feature/mailchimp

### LINKED ISSUE #11: Integrate mailchimp for people to subscribe as testers
https://www.raymondcamden.com/2019/05/29/using-the-mailchimp-api-with-netlify-serverless-functions

### PR DESCRIPTION
**Pull request by yash296**
Mailchimp testing pending

### CODE REVIEW

File: src/components/footer.vue [hunk 1]
```diff
@@ -62,17 +96,47 @@
...
-    emailIsValid: false
+    snackbar: false,
+    message: "",
+    emailIsValid: false,
   }),
   methods: {
+    async submitEmail() {
+      await axios
+        .get(
+          "https://www.nexd.app/.netlify/functions/mailchimp?email=" +
```
Comment (f10l) [inline src/components/footer.vue]:
```suggestion
          "/.netlify/functions/mailchimp?email=" +
```
You can simply use this on the current host, which will allow to work then also for previews and other domains.

**Review by f10l — COMMENTED** [review]
Looks good so far, I will need another test after resolving the comment.

File: src/components/footer.vue [hunk 2]
```diff
@@ -62,17 +96,47 @@
...
+    async submitEmail() {
+      await axios
+        .get(
+          "/.netlify/functions/mailchimp?email=" +
+            this.email
+        )
+        .then((response) => {
+          if (
+            response &&
+            response.statusCode &&
```
Comment (f10l) [inline src/components/footer.vue]:
```suggestion
```
This line is unnecessary.

File: src/components/footer.vue [hunk 3]
```diff
@@ -62,17 +96,47 @@
...
+            response.statusCode === 200 &&
+            response.body &&
+            response.body.status &&
+            response.body.status === "subscribed"
```
Comment (f10l) [inline src/components/footer.vue]:
```suggestion
            response &&
            response.status === "subscribed"
```
I think it is not nested in the body property.

**Review by f10l — COMMENTED** [review]
The confirmation did not work for me, but it inserted in the mailing list 👍 
See comments.

**Comment by yash296** [comment]
Made the required changes

File: src/components/footer.vue [hunk 4]
```diff
@@ -62,17 +96,41 @@
...
   methods: {
+    async submitEmail() {
+      await axios
+        .get("/.netlify/functions/mailchimp?email=" + this.email)
+        .then((response) => {
+          if (response && response.status) {
+            if (response.status === "subscribed") {
+              this.message =
+                "Sie haben sich jetzt als Tester für Nexd registriert";
+            } else if (response.status === 400) {
```
Comment (f10l) [inline src/components/footer.vue:116]:
```suggestion
            } else if (response.statusCode === 400) {
```

**Review by f10l — COMMENTED** [review]
I do not find the problem :( 
The request looks fine to me and also the `response.status` is `subscribed`.
Bad still somehow, I get the "Ups! Irgendwas lief schief". Does the subscription work for you?

**Pull request merged**